### ISSUE TITLE
Hang when disposing NetMQMonitor after TerminatingException

### ISSUE THREAD

**Issue by drew-r** [zeromq/netmq#212]
I'm finding that a `NetMQMonitor` I'm using to detect disconnects, is hanging indefinitely when trying to Dispose if it encounters a `TerminatingException` after calling `.Start()`.

I'm working under .NET 3.5 but I see this in 4.5 also.

Code sample below:

```csharp 
 using (NetMQMonitor monitor = new NetMQMonitor(_context, socket, "inproc://monitor", SocketEvent.Disconnected))
                {
                    monitor.Disconnected += (s, e) =>
                    {
                        disconnectCallback();
                    };
                    try
                    {
                        monitor.Start();
                    }
                    catch (TerminatingException) 
                    { 
                        Trace.WriteLine("socketMonitor caught TerminatingException.");
                     }
                }    // seems to hang here
                Trace.WriteLine("socketMonitor disposed.");
```

It's monitoring a PULL socket. Any ideas?

**Comment by somdoron** [comment]
@drew-r I think this was already fixed in #206, can you please compile from repository and check?

**Comment by drew-r** [comment]
@somdoron this is fixed in the repo. thanks
